internal/commands/inspect_builder.go

```diff
@@ -266,35 +272,152 @@ func runImagesOutput(runImage string, mirrors []string, cfg config.Config) (stri
 	return strings.TrimSuffix(buf.String(), "\n"), nil
 }
 
-func detectionOrderOutput(order dist.Order) (string, error) {
+type stackEntry struct {
+	LayerInfo dist.BuildpackRef
+	Depth     int
+	Last      bool
+}
+
+func reverseStack(stack []stackEntry) []stackEntry {
+	left := 0
+	right := len(stack) - 1
+	for left < right {
+		stack[left], stack[right] = stack[right], stack[left]
+		left++
+		right--
+	}
+	return stack
+}
+
+func detectionOrderOutput(order dist.Order, layers dist.BuildpackLayers, maxDepth int) (string, error) {
```

Comment:
nit: This looks like an intimidatingly long `func`. Any chance we can decompose this into smaller more easily digestible `funcs`?

Reply:
Pulled out some of the messier logic, let me know if you think there is more to be done around this.

Reply:
Looks much better!